packages/synchro-charts/src/components/sc-dial/sc-dial-base/sc-dial-base.tsx

```diff
@@ -0,0 +1,94 @@
+import { Component, h, Prop, State } from '@stencil/core';
+import { DataPoint, DataStream, Primitive, SizeConfig, ViewPortConfig } from '../../../utils/dataTypes';
+import { isNumberDataStream } from '../../../utils/predicates';
+import { Threshold } from '../../charts/common/types';
+import { DialLoading } from './sc-dial-loading';
+import { TextSizeConfig, sizeContent, sizeConfigurations } from './util';
+
+const title = (dataStream: { detailedName?: any; name?: any } | null | false) => {
+  if (dataStream) {
+    return dataStream.detailedName || dataStream.name;
+  }
+  return null;
+};
+
+@Component({
+  tag: 'sc-dial-base',
+  styleUrl: 'sc-dial-base.css',
+  shadow: false,
+})
+export class ScDialBase {
+  @Prop() viewport: ViewPortConfig;
+  @Prop() breachedThreshold?: Threshold;
+  @Prop() valueColor?: string; // css color string
+
+  @Prop() alarmStream?: DataStream;
+
+  @Prop() propertyStream?: DataStream;
+  @Prop() propertyPoint?: DataPoint<Primitive>;
+  @Prop() size?: SizeConfig & { fontSize?: string };
```

Comment:
not quite the API we want for size, which should be a string what you can specify things such as just `size="XS"`

Reply:
Done --- Add size = "XS" "S" "M" "L" "XL" and "XXL", six kinds of size. Their size is set according to emaill requirements.